sdk/storage_blobs/src/clients/container_client.rs

```diff
@@ -15,30 +19,83 @@ use azure_storage::{
 };
 use time::OffsetDateTime;
 
-pub trait AsContainerClient {
-    fn container_client(&self, container_name: impl Into<String>) -> ContainerClient;
+#[derive(Clone, Debug)]
+pub struct ContainerClientBuilder {
+    storage_credentials: Option<StorageCredentials>,
+    storage_account: String,
+    container: String,
+    storage_options: StorageOptions,
 }
 
-impl AsContainerClient for StorageClient {
-    fn container_client(&self, container_name: impl Into<String>) -> ContainerClient {
-        ContainerClient::new(self.clone(), container_name.into())
+impl ContainerClientBuilder {
+    // TODO: I actually think it would be best to have the credentails as part of this signature, which would make them "required"
```

Comment:
I agree. This would mirror how cosmos handles this `CosmosClientBuilder::new` takes `account` and `auth_token` (which is essentially the spirtual equivalent to `StorageCredentials`).